src/pbt/__init__.py

```diff
@@ -30,104 +42,239 @@ def __init__(self, path_root: str):
+                    if os.path.isfile(path_pipeline_build_absolute) and path_pipeline_build.endswith('.whl'):
+                        build_file_found = True
+                        self.pipelines_build_path[path_pipeline] = {
+                            'source_absolute': path_pipeline_build_absolute,
+                            'source': path_pipeline_build,
+                            'uploaded': False
+                        }
+            elif self.project_language == 'scala':
+                path_pipeline_dist = path_pipeline_absolute + '/target'
+                for path_pipeline_build in os.listdir(path_pipeline_dist):
+                    path_pipeline_build_absolute = path_pipeline_dist + '/' + path_pipeline_build
+                    if os.path.isfile(path_pipeline_build_absolute) and path_pipeline_build.endswith('.jar'):
+                        build_file_found = True
+                        self.pipelines_build_path[path_pipeline] = {
+                            'source_absolute': path_pipeline_build_absolute,
+                            'source': path_pipeline_build,
+                            'uploaded': False
+                        }
+
+            if rc == 0:
+                if build_file_found:
+                    print("\n[bold blue]✅ Build complete![/bold blue]")
+                else:
+                    print(
+                        f"\n[bold red]❌ Build completed but built target not found for pipeline: {pipeline['name']}![/bold red]")
+                    overall_build_status = False
+            else:
+                print(f"\n[bold red]❌ Build failed for pipeline: {pipeline['name']}![/bold red]")
+                overall_build_status = False
+
+        if not overall_build_status:
+            sys.exit(1)
 
     def deploy(self):
         self.build()
 
         print('\n[bold blue]Deploying %s jobs ⏱[/bold blue]' % self.jobs_count)
+
+        pipelines_upload_failures = collections.defaultdict(list)
+        job_update_failures = dict()
+
         for job_idx, (path_job, job) in enumerate(self.jobs.items()):
             print('\n  Deploying job %s [%s/%s]' % (path_job, job_idx + 1, self.jobs_count))
 
             path_job_absolute = os.path.join(os.path.join(self.path_root, path_job), 'code')
             path_job_definition = os.path.join(path_job_absolute, 'databricks-job.json')
 
-            job_definition = {}
             with open(path_job_definition, 'r') as _in:
                 job_definition = json.load(_in)
 
             components = job_definition['components']
+
             for component in components:
                 if 'PipelineComponent' in component:
                     pipeline_uri = component['PipelineComponent']['id']
 
-                    uri_pattern = '([0-9]+)/([-_.A-Za-z0-9 /]+)'
+                    # Matches project_id/pipelines/pipeline_name or pipelines/pipeline_name or /pipelines/pipeline_name
```

Comment:
`'([0-9]*)(pipelines/[-_.A-Za-z0-9 /]+)'` does not match `/pipelines/pipeline_name` actually. Either change the regex, or change the comment. Let's keep it consistent.

Reply:
Updated comment for now. As of now that pattern wont be needed